src/Docker.DotNet/DockerClient.cs

```diff
@@ -121,6 +121,7 @@ internal DockerClient(DockerClientConfiguration configuration, Version requested
             _endpointBaseUri = uri;
 
             _client = new HttpClient(Configuration.Credentials.GetHandler(handler), true);
+            _client.DefaultRequestHeaders.Add(nameof(_endpointBaseUri.Host), _endpointBaseUri.Host);
```

Comment:
Was this a bug? Why do we need it?

Reply:
At some point test were failing because there was no Host header within request. it seems to work fine now. Reverting in next commit